src/Microsoft.VisualStudio.Threading/JoinableTaskContext.cs

```diff
@@ -302,11 +302,6 @@ protected internal virtual SynchronizationContext NoMessagePumpSynchronizationCo
     {
         get
         {
-            // Callers of this method are about to take a private lock, which tends
```

Comment:
You could avoid the debugger hangs by adding `[DebuggerBrowsable(DebuggerBrowsableState.Never)]` to the property.